### PR TITLE
- Fix for reconnection to Slack server and not dropping messages.

### PR DESCRIPTION
**Pull request by souravran**
My idea here is to not drop any messages due to connection loss to the slack server. In this PR, along with reconnection functionality, I have added a buffer that would grab all the subscribed messages in case there is no established connection. Once connection resumes, stored messages will be sent to the slack channel.

### CODE REVIEW

File: slack_ros/scripts/slack_ros.py [hunk 1]
```diff
@@ -34,46 +34,72 @@ def __init__(self):
...
+        self.pub = rospy.Publisher('from_slack_to_ros', String, queue_size=10)
+        rospy.Subscriber("from_ros_to_slack", String, self.message_callback)
+        rospy.Subscriber("send_file_to_slack", String, self.file_callback)
 
-	# Create the slack client
-	self.sc = SlackClient(self.token)
+        # Create the slack client
+        self.slack_client = SlackClient(self.token)
+        self.has_connection = False
+        self.subscribed_msg_store = []
```
Comment (floweisshardt) [inline slack_ros/scripts/slack_ros.py:44]: there should be no need to save messages in a queue ourselves. ROS subscriber (with queue size) will handle that for us. we just have to block the callback method until we've reconnected to the slack service. ROS will not execute callbacks in parallel but sequentially as soon as the subscriber callback finished. Thus this is exactly the behaviour we're looking for.

Please use a configurable queue size and remove ```subscribed_msg_store```

**Review by floweisshardt — CHANGES_REQUESTED** [review]

File: slack_ros/scripts/slack_ros.py (hunk 1 above)
Comment (souravran) [reply slack_ros/scripts/slack_ros.py:44]: My initial idea was exactly the same as it has been described above, but I could not find a way to block the callback. One way is to unsubscribe/resubscribe and that's not really blocking the callback, and again whether that is efficient is questionable. Or use semaphore inside the callback method itself, I have not tried this and don't know if threads would go to sleep while blocking the callback.

Any idea is welcome, meanwhile I would further look into this and update the PR.

Comment (benmaidel) [reply slack_ros/scripts/slack_ros.py:44]: You could use a condition variable for this. Just wait/block inside the topic callback until the condition is fulfilled (connected to slack). https://docs.python.org/2/library/threading.html
With this kind of implementation you need some kind of connectivity check routine.

On the other side. Creating a separate queue like you did isn't that wrong. As long as there are elements within the queue you can try sending them. If an error during sending happens you can try reconnect and leave the element inside the queue as long as the connection problems appears...

@ipa-fmw what do you think?

**Comment by fmessmer** [comment]
wont merge...this package is no longer used

**Pull request closed without merging**